resources/codegen/tests/test___main__.py

```diff
@@ -0,0 +1,202 @@
+        )
+
+    def test_when_set_then_copyright_added_with_owner(self, tmp_path, yaml_path_minimal):
+        execute_cli("--python", "--copyright-owner", "An Owner", "-o", tmp_path, yaml_path_minimal)
+
+        contents = file_contents(tmp_path, "python", "p_p", "__init__.py")
+        assert "www.apache.org/licenses/LICENSE-2.0" in contents
+        assert "An Owner" in contents
+
+
+class Test_CLI_args_copyright_date:
+    def test_when_not_set_then_copyright_date_is_this_year(self, tmp_path, yaml_path_minimal):
+        execute_cli("--python", "--copyright-owner", "An Owner", "-o", tmp_path, yaml_path_minimal)
+
+        contents = file_contents(tmp_path, "python", "p_p", "__init__.py")
+
+        assert "www.apache.org/licenses/LICENSE-2.0" in contents
+        assert f"Copyright {datetime.date.today().year} An Owner" in contents
+
+    def test_when_set_then_copyright_date_is_as_specified(self, tmp_path, yaml_path_minimal):
+        execute_cli(
+            "--python",
+            "--copyright-owner",
+            "An Owner",
+            "--copyright-date",
+            "2010-2022",
+            "-o",
+            tmp_path,
+            yaml_path_minimal,
+        )
+        contents = file_contents(tmp_path, "python", "p_p", "__init__.py")
+
+        assert "www.apache.org/licenses/LICENSE-2.0" in contents
+        assert "Copyright 2010-2022 An Owner" in contents
+
+
+class Test_CLI_args_help:
+    def test_when_h_set_then_help_is_generated(self, tmp_path, yaml_path_minimal):
+        assert "usage: openassetio-codegen" in execute_cli("-h").stdout
+
+    def test_when_help_set_then_help_is_generated(self, tmp_path, yaml_path_minimal):
+        assert "usage: openassetio-codegen" in execute_cli("--help").stdout
+
+
+class Test_CLI_args_python:
+    def test_when_python_set_then_python_is_generated(self, tmp_path, yaml_path_minimal):
+        execute_cli("--python", "-o", tmp_path, yaml_path_minimal)
+
+        assert os.path.isdir(os.path.join(tmp_path, "python"))
+
+    def test_when_only_python_set_then_only_python_is_generated(self, tmp_path, yaml_path_minimal):
+        execute_cli("--python", "-o", tmp_path, yaml_path_minimal)
+
+        assert os.listdir(tmp_path) == [
+            "python",
+        ]
+
+
+class Test_CLI_default_languages:
+    def test_when_none_set_then_all_languages_are_generated(self, tmp_path, yaml_path_minimal):
+        execute_cli("--python", "-o", tmp_path, yaml_path_minimal)
+
+        assert os.listdir(tmp_path) == [
+            "python",
+        ]
+
+
+#
+# Helpers
+#
+
+
+def execute_cli(*args):
+    all_args = ["openassetio-codegen"]
+    all_args.extend(args)
+    # We explicitly don't want an exception to be raised.
+    # pylint: disable=subprocess-run-check
+    return subprocess.run(all_args, capture_output=True, encoding="utf-8")
+
+
+def file_contents(*args):
+    path = os.path.join(*args)
+    return "\n".join(open(path, encoding="utf=8").readlines())
```

Comment:
[fix] Kinda surprised this works, since "utf=8" isn't mentioned in [the docs](https://docs.python.org/3/library/codecs.html#standard-encodings) (it mentions case insensitivity and `-` and `_` being interchangeable, but not `=`).
```suggestion
    return "\n".join(open(path, encoding="utf-8").readlines())
```

Reply:
They love their `-` to `_`!!!  Great spot

Reply:
fixed [here](https://github.com/OpenAssetIO/OpenAssetIO/pull/529/commits/5e2f4fad94e01dee20cf691d37e8d254d0a45154).

Reply:
I believe the correct commit is 246605cf :+1: